### ISSUE TITLE
It's not possible to set regulation mode to Expert using Automations

### ISSUE THREAD

**Issue by Edwintenhaaf** [jmcollin78/versatile_thermostat#717]
It's not possible to set regulation mode to Expert using Automations
using version: 6.8.3
HA: 2024.12.4

<img width="1250" alt="image" src="https://github.com/user-attachments/assets/97a25912-3b72-4aa7-8ef6-7b8429b42ff7" />

`Failed to perform the action versatile_thermostat.set_auto_regulation_mode. value must be one of ['Light', 'Medium', 'None', 'Slow', 'Strong'] for dictionary value @ data['auto_regulation_mode']. Got None`

I have a working Expert mode in my configuration

**Comment by jmcollin78** [comment]
Hello,

You are right and it is easy to fix. I'm curious of your use case. Why do you need to change the regulation level to expert via an automation ?

**Comment by jmcollin78** [comment]
This is a very easy PR if someone wants to try !

**Comment by Edwintenhaaf** [comment]
Hi @jmcollin78, yes I will try to fix it myself.

**Comment by Edwintenhaaf** [comment]
> Hello,
> 
> You are right and it is easy to fix. I'm curious of your use case. Why do you need to change the regulation level to expert via an automation ?

Because I want to smoothen the spikes I got in temperature when the outside weather -temperature, wind- changes.
So I created some thresholds based on historic data.
![image](https://github.com/user-attachments/assets/2bb0633e-183f-41b9-baa1-50a23ae4dc28)
```
      - alias: Regulation mode to strong
        if:
          - condition: numeric_state
            entity_id: sensor.outdoor_temperature
            below: 5
        then:
          - action: versatile_thermostat.set_auto_regulation_mode
            metadata: {}
            data:
              auto_regulation_mode: Strong
            target:
              entity_id: climate.woonkamer
        enabled: true
      - alias: Regulation mode to medium
        if:
          - condition: numeric_state
            entity_id: sensor.outdoor_temperature
            above: 5
          - condition: numeric_state
            entity_id: sensor.outdoor_temperature
            below: 12
        then:
          - action: versatile_thermostat.set_auto_regulation_mode
            metadata: {}
            data:
              auto_regulation_mode: Medium
            target:
              entity_id: climate.woonkamer
        enabled: true
      - alias: Regulation mode to light
        if:
          - condition: numeric_state
            entity_id: sensor.outdoor_temperature
            above: 12
        then:
          - action: versatile_thermostat.set_auto_regulation_mode
            metadata: {}
            data:
              auto_regulation_mode: Light
            target:
              entity_id: climate.woonkamer
```
When the temperature drops below 5 I want to switch to expert mode.

```
versatile_thermostat:
      auto_regulation_expert:
          kp: 0.5
          ki: 0.1
          k_ext: 0.15
          offset_max: 2
          stabilization_threshold: 0.1
          accumulated_error_threshold: 10
```

There must be other ways but didn't find out how.

**Comment by jmcollin78** [comment]
Huum that is interesting but i'm not sure it is useful. If sometimes you need a very strong regulation (with expert) you should always stay in that mode.
Level of regulation is just a maximum level so it can do a light if only light regulation is needed.

 Have you tried this ?

**Comment by Edwintenhaaf** [comment]
> Huum that is interesting but i'm not sure it is useful. If sometimes you need a very strong regulation (with expert) you should always stay in that mode. Level of regulation is just a maximum level so it can do a light if only light regulation is needed.
> 
> Have you tried this ?

Yes I tried several things including a custom fan mode automation 
Problem -I think- is the placement of the ac unit in a corner of the living room.
Temperature rises in the corner but does not reach the other side of the room. Or it's allready cooled down.
The 'use temperature of underlying' together with custom fan speed automation solves this for the most.
But sometimes the temperature starts jumping up and down a seen in the graph I posted earlier.

With a light regulation and cold temperatures outside it never reaches the target temp.
Central heating is also part of the automation so that kicks in when temperature drops as a 'safety measure'

during the christmas days i am free so i have all the time to optimize the system further. Wouldn't it be nice if everything worked perfectly right ;-)
